### ISSUE TITLE
Parsing 64 bit integer returns 32 bit ones

### ISSUE THREAD

**Issue by smeng9** [AlexAltea/libelf.js#3]
Hi,

I have tried to use the demo https://alexaltea.github.io/libelf.js/ to parse some of 32bit elf binaries.

However I got the wrong entry address in the header, seems it is missing 0x0804 prefix.

I have traced the issue and seems __read_i64 function is returning 32 bit values at https://github.com/AlexAltea/libelf.js/blob/master/src/libelf-wrapper.js#L70
Could you please take a look? Thanks

**Comment by AlexAltea** [comment]
> I have traced the issue and seems __read_i64 function is returning 32 bit values at https://github.com/AlexAltea/libelf.js/blob/master/src/libelf-wrapper.js#L70

Oh that last argument should be changed to *64*:

```diff
-            MLibelf.getValue(addr+4, 'i32')], 32);
+            MLibelf.getValue(addr+4, 'i32')], 64);
```

Can you test and see if it works?

**Comment by smeng9** [comment]
I think that may not be the only issue and I still did not get it to work, actually the entry_address I got is even truncated to lower 16 bits.
https://github.com/AlexAltea/libelf.js/blob/master/src/libelf-wrapper.js#L185

I have further traced the issue to the following minimum case that can reproduce:
a = new ElfUInt32([4198484])
a.hex() // 0x00001054, which should be 0x00401054, the higher bits should be kept for 32 bit integer

The ElfUInt class may be not implemented correctly when casting integer using this.__integer and we can add some test cases.

Thank you.

**Comment by AlexAltea** [comment]
Wow, I just checked the *libelf-integers.js* and indeed, it's buggy and with critical features left unimplemented.
My apologies about that... I'll try to provide a fix for this class in the coming days.

**Comment by AlexAltea** [comment]
@smeng9 I've fixed these issues in 385725567ae2c9f1d00c3408def13e1b21b6c5af + 4fc9a07d53291ad544475abb3be476691aa6640c, and added some tests in 5f548fe85056a20bb2ad6dbe621271ca374685ad.
Can you check if it works now?

EDIT: `libelf.min.js` updated as of 9aeb9a104d58f078e1c3d66fb45ffe6e713bf3eb.

**Comment by smeng9** [comment]
Now the entry address looks correct on https://alexaltea.github.io/libelf.js/

Thank you so much, I can close the issue now
